File: java/0242_isAnagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(t.length()!=s.length())
            return false;
        int[] arr = new int[26];
        for(var i=0;i<s.length();i++)
            arr[s.charAt(i)-'a']++;
        for(var i=0;i<t.length();i++)
        {
            arr[t.charAt(i)-'a']--;
            if(arr[t.charAt(i)-'a']<0)
                return false;
        }
        return true;
    }
}

File: cpp/0242_isAnagram.cpp
// t 是 s 的异位词等价于「两个字符串排序后相等」
// 先建立一个hashmap，遍历t得到char-freq，再遍历t减去freq
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size())
            return false;
        vector<int> arr(26,0);
        
        for(auto &c:s)
            arr[c-'a']++;
        
        for(auto &c:t)
        {
            arr[c-'a']--;
            if(arr[c-'a']<0)
                return false;
        }
        return true;
    }
};
